packages/http-client-java/emitter/src/code-model-builder.ts

```diff
@@ -604,48 +618,51 @@ export class CodeModelBuilder {
           codeModelClient.operationGroups.push(codeModelGroup);
         }
       }
-      this.codeModel.clients.push(codeModelClient);
+    }
+
+    this.codeModel.clients.push(codeModelClient);
 
-      // postprocess for ServiceVersion
-      let apiVersionSameForAllClients = true;
-      let sharedApiVersions = undefined;
+    // postprocess for ServiceVersion
+    let apiVersionSameForAllClients = true;
+    let sharedApiVersions = undefined;
+    for (const client of this.codeModel.clients) {
+      const apiVersions = client.apiVersions;
+      if (!apiVersions) {
+        // client does not have apiVersions
+        apiVersionSameForAllClients = false;
+      } else if (!sharedApiVersions) {
+        // first client, set it to sharedApiVersions
+        sharedApiVersions = apiVersions;
+      } else {
+        apiVersionSameForAllClients = isEqual(sharedApiVersions, apiVersions);
+      }
+      if (!apiVersionSameForAllClients) {
+        break;
+      }
+    }
+    if (apiVersionSameForAllClients) {
+      const serviceVersion = getServiceVersion(this.codeModel);
       for (const client of this.codeModel.clients) {
-        const apiVersions = client.apiVersions;
-        if (!apiVersions) {
-          // client does not have apiVersions
-          apiVersionSameForAllClients = false;
-        } else if (!sharedApiVersions) {
-          // first client, set it to sharedApiVersions
-          sharedApiVersions = apiVersions;
-        } else {
-          apiVersionSameForAllClients = isEqual(sharedApiVersions, apiVersions);
-        }
-        if (!apiVersionSameForAllClients) {
-          break;
-        }
+        client.serviceVersion = serviceVersion;
       }
-      if (apiVersionSameForAllClients) {
-        const serviceVersion = getServiceVersion(this.codeModel);
-        for (const client of this.codeModel.clients) {
-          client.serviceVersion = serviceVersion;
-        }
-      } else {
-        for (const client of this.codeModel.clients) {
-          const apiVersions = client.apiVersions;
-          if (apiVersions) {
-            client.serviceVersion = getServiceVersion(client);
-          }
+    } else {
+      for (const client of this.codeModel.clients) {
+        const apiVersions = client.apiVersions;
+        if (apiVersions) {
+          client.serviceVersion = getServiceVersion(client);
         }
       }
     }
+
+    return codeModelClient;
   }
 
   private listSubClientsUnderClient(
     client: SdkClientType<SdkHttpOperation>,
-    includeNestedOperationGroups: boolean,
-    isRootClient: boolean,
```

Comment:
This param seems not needed. If a client does not have parent, it is the root client.